pkg/ccl/backupccl/backup_telemetry.go

```diff
@@ -0,0 +1,486 @@
+		recoveryEvent.AuthTypes = append(recoveryEvent.AuthTypes, typ)
+	}
+	sort.Strings(recoveryEvent.AuthTypes)
+	for typ := range storageTypes {
+		recoveryEvent.DestinationStorageTypes = append(recoveryEvent.DestinationStorageTypes, typ)
+	}
+	sort.Strings(recoveryEvent.DestinationStorageTypes)
+
+	log.StructuredEvent(ctx, event)
+}
+
+func getLargestScope(
+	coverage tree.DescriptorCoverage, requestedDescriptors []descpb.Descriptor,
+) targetScope {
+	largestScope := unknownScope
+
+	if coverage == tree.AllDescriptors {
+		largestScope = clusterScope
+	} else {
+		// Log the largest scope from the targets.
+		for i := range requestedDescriptors {
+			var scope targetScope
+			tbl, db, _, sc := descpb.FromDescriptor(&requestedDescriptors[i])
+			if tbl != nil {
+				scope = tableScope
+			} else if sc != nil {
+				scope = schemaScope
+			} else if db != nil {
+				scope = databaseScope
+			}
+
+			if scope > largestScope {
+				largestScope = scope
+			}
+		}
+	}
+	return largestScope
+}
+
+func parseStorageAndAuth(uri string) (string, string, error) {
+	var storageType string
+	authType := cloud.AuthParamSpecified
+	parsedURI, err := url.Parse(uri)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "failed to parse URI %s for telemetry", uri)
+	}
+	storageType = parsedURI.Scheme
+	auth := parsedURI.Query().Get(cloud.AuthParam)
+	if auth != "" {
+		authType = auth
+	}
+	return storageType, authType, nil
+}
+
+func loggedWaitBehavior(b jobspb.ScheduleDetails_WaitBehavior) string {
+	switch b {
+	case jobspb.ScheduleDetails_NO_WAIT:
+		return "no-wait"
+	case jobspb.ScheduleDetails_SKIP:
+		return "skip"
+	default:
+		return "wait"
+	}
+}
+
+func loggedErrorHandlingBehavior(b jobspb.ScheduleDetails_ErrorHandlingBehavior) string {
+	switch b {
+	case jobspb.ScheduleDetails_RETRY_SOON:
+		return "retry-soon"
+	case jobspb.ScheduleDetails_PAUSE_SCHED:
+		return "pause-schedule"
+	default:
+		return "retry-schedule"
+	}
+}
+
+func loggedSubdirType(subdir string) string {
+	timeBaseSubdir := true
+	var subdirType string
+	if _, err := time.Parse(DateBasedIntoFolderName,
+		subdir); err != nil {
+		timeBaseSubdir = false
+	}
+
+	if strings.EqualFold(subdir, latestFileName) {
+		subdirType = latestSubdirType
+	} else if !timeBaseSubdir {
+		subdirType = customSubdirType
+	} else {
+		subdirType = standardSubdirType
+	}
+
+	return subdirType
+}
+
+// logCreateScheduleTelemetry publishes an eventpb.RecoveryEvent about a created
+// backup schedule.
+func logCreateScheduleTelemetry(
+	ctx context.Context,
+	incRecurrence *scheduleRecurrence,
+	fullRecurrence *scheduleRecurrence,
+	incScheduledJob *jobs.ScheduledJob,
+	fullScheduledJob *jobs.ScheduledJob,
+	firstRun *time.Time,
+	ignoreExisting bool,
+	details jobspb.ScheduleDetails,
+) {
+	var event eventpb.EventPayload
+
+	var firstRunStr string
+	if firstRun != nil {
+		firstRunStr = firstRun.Format(time.RFC3339)
+	}
+
+	// TODO: add rest of the actual backup logging here.
```

Comment:
Can we say a bit more here? Some of the logging we do for backups we can't do until we do resolution from the looks of it.

Reply:
> I haven't really figured out a way to log the resolution of the actual backup at the time of schedule creation

Above where we log this telemetry, we actually run the planning hook which constructs the backupDetails from the backupNode, and based on what we are currently collecting, I think you would be able to get most if not all of the telemetry we collect for backups; but I may be misunderstanding.

But, if we are still going to collect the full telemetry record for each backup that this schedule triggers, I'm not sure it is really worth it.  Is there a big value on the querying side of the data being flatter?  

If we are concerned about the jobs from these schedules sending us the same telemetry every day and wanted to elide the telemetry in the case of it originating from a schedule, that could be interesting.